src/main/java/com/shaft/tools/io/PdfFileManager.java

```diff
@@ -0,0 +1,173 @@
+			parser.parse();
+		} catch (IOException e) {
+			ReportManagerHelper.log(e);
+			ReportManager.log(
+					"Couldn't parse the stream that opened the document to be prepared to populate the COSDocument object.");
+			Assert.fail(
+					"Couldn't parse the stream that opened the document to be prepared to populate the COSDocument object.");
+		}
+		return parser;
+	}
+
+	private COSDocument getParsedDocument(PDFParser parser) {
+		try {
+			cosDoc = parser.getDocument();
+		} catch (IOException e) {
+			ReportManagerHelper.log(e);
+			ReportManager.log(
+					"Couldn't get the document that was parsed. Check that the document parsed before get the document.");
+			Assert.fail(
+					"Couldn't get the document that was parsed. Check that the document parsed before get the document.");
+		}
+		return cosDoc;
+	}
+
+	private String getPdfText(COSDocument cosDoc, int startPageNumber, int endPageNumber) {
+		try {
+			strip = new PDFTextStripper();
+			// By default, text extraction is done in the same sequence as the text in the
+			// PDF page content stream. PDF is a graphic format, not a text format, and
+			// unlike HTML, it has no requirements that text one on page be rendered in a
+			// certain order. The order is the one that was determined by the software that
+			// created the PDF
+			// To get text sorted from left to right and top to bottom
+			strip.setSortByPosition(true);
+		} catch (IOException e) {
+			ReportManagerHelper.log(e);
+			ReportManager.log("Couldn't load PDFTextStripper properties.");
+			Assert.fail("Couldn't load PDFTextStripper properties.");
+		}
+
+		strip.setStartPage(startPageNumber);
+		strip.setEndPage(endPageNumber);
+		PDDocument pdDoc = new PDDocument(cosDoc);
+
+		String content = null;
+		try {
+			content = strip.getText(pdDoc);
+		} catch (IOException e) {
+			ReportManagerHelper.log(e);
+			ReportManager.log("Couldn't get document text. Document state is invalid or it is encrypted.");
+			Assert.fail("Couldn't get document text. Document state is invalid or it is encrypted.");
+		}
+		return content;
+	}
+
+	private void closeStreamAndDeleteFile(File file, RandomAccessBufferedFileInputStream stream,
+			DeleteFileAfterValidationStatus deleteFileAfterValidation) {
+		try {
+			stream.close();
+		} catch (IOException e) {
+			ReportManagerHelper.log(e);
+			ReportManager.log("Couldn't close the stream, check if it already opened.");
+			Assert.fail("Couldn't close the stream, check if it already opened.");
+		}
+
+		// Delete the file from target folder for next run
+
+		switch (deleteFileAfterValidation) {
+		case TRUE:
+			try {
+				FileUtils.forceDelete(file);
+			} catch (IOException e) {
+				ReportManagerHelper.log(e);
+				ReportManager.log("Couldn't find the file, File directory may be null or file is not found.");
+				Assert.fail("Couldn't find the file, File directory may be null or file is not found.");
+			}
+		default:
+			break;
+		}
+
+	}
+
+	// Will implement same for preview pdf file
```

Comment:
Add a todo to create a new validation inside the file validations wizard that compares the content of the downloaded file, versus the content of a test data file and attaches both of them in the report.

I can do that one to complement the feature you're working on. Or I can help you do it if you prefer.

Reply:
@MohabMohie yes please if you can help in that, I'd like to do it.

Reply:
@MamdouhMansour great, let's finalize this PR and then we can open a new one for the validation that will use this method (Y)

Reply:
@MohabMohie I am done and waiting for your comments, thank you for the support as usual